### ISSUE TITLE
`conda config --show` doesn't show all channels

### ISSUE THREAD

**Issue by pitrou** [conda/conda#3302]
```
(35) Z:\numba>conda config --get channels
--add channels 'defaults'   # lowest priority
--add channels 'numba'
--add channels 'antoine'   # highest priority

(35) Z:\numba>conda config --show
channel_alias: https://conda.anaconda.org/
shortcuts: true
changeps1: true
always_copy: false
use_pip: true
always_yes: false
disallow: []
default_channels:
  - https://repo.continuum.io/pkgs/free
  - https://repo.continuum.io/pkgs/pro
  - https://repo.continuum.io/pkgs/msys2
create_default_packages: []
ssl_verify: true
offline: false
add_anaconda_token: true
add_pip_as_python_dependency: true
track_features: []
channels:
  - defaults
proxy_servers: {}
quiet: false
json: false
auto_update_conda: true
binstar_upload:
update_dependencies: true
allow_softlinks: true
debug: false
channel_priority: true
show_channel_urls:

(35) Z:\numba>conda info
Current conda install:

               platform : win-64
          conda version : 4.2.1
       conda is private : False
      conda-env version : 2.5.2
    conda-build version : 1.21.11+0.g5b44ab3.dirty
         python version : 3.5.1.final.0
       requests version : 2.10.0
       root environment : C:\Miniconda3  (writable)
    default environment : C:\Miniconda3\envs\35
       envs directories : C:\Miniconda3\envs
          package cache : C:\Miniconda3\pkgs
           channel URLs : defaults
            config file : C:\Users\Antoine\.condarc
           offline mode : False
```

**Comment by HugoTian** [comment]
Wired, I cannot reproduce this error. 

But I got another issue, the output format is not stable. The label changes position from time to time.

Moreover, please refer to this PR #3304 ,  when we do conda config --get channels, the top one should have highest priority. Wrong output for conda config --get channel

**Comment by HugoTian** [comment]
@pitrou  Can you show me the condarc file , as well as the path of condarc file ?

**Comment by pitrou** [comment]
The condarc file is at `C:\Users\Antoine\.condarc`. Its contents are the following:
```
channels:
  - antoine
  - numba
  - defaults
```

**Comment by kalefranz** [comment]
There was a bug in 4.2.1 where conda didn't pick home the condarc file in the home directory on Windows.  Fixed in #3228.  Please update to 4.2.3 :)
